src/main/java/chess/tools/Chess.java

```diff
@@ -0,0 +1,149 @@
+package chess.tools;
+
+import chess.tools.game.*;
+import chess.tools.model.ChessModel;
+import chess.tools.move.MoveManager;
+import chess.tools.move.MoveTuple;
+import chess.tools.move.Position;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+public class Chess {
+
+    private ChessModel chessModel;
+    private MoveManager moveManager;
+    private Logger logger = Logger.getLogger(Chess.class.getName());
+    private CommandParser parser = new CommandParser();
+    private final GameEndCalculator endCalculator = new GameEndCalculator();
+
+    public static final PositionMapping MAPPING = new PositionMapping();
+
+    public Chess() {
+        chessModel = new ChessModel();
+        moveManager = new MoveManager(chessModel);
+    }
+
+    public Chess(List<String> terms) {
+        chessModel = new ChessModel();
+        moveManager = new MoveManager(chessModel);
+        terms.forEach(this::processTerm);
+    }
+
+    /**
+     * @param move - a possible move
+     * @return valid or not
+     */
+    public boolean gameStep(MoveTuple move) {
+        final int sizeBefore = chessModel.getTurnValids().size();
+        processMove(move);
+        int size = chessModel.getTurnValids().size();
+        return (size == sizeBefore + 1) && chessModel.getTurnValids().get(size - 1);
+    }
+
+    public boolean colorValidation(Position begin) {
+        Figure oldBegin = chessModel.getFigureOnBoard(begin);
+        ChessColor color = oldBegin.getColor();
+        return color.validateTurn(moveManager.getTurnCounter());
+    }
+
+    public void processTerm(String term) {
+        logger.info("Do move: " + term);
+        endCalculatorChecks();
+        if (turn(term) && !isStalement()) {
+            calculateCheckMate();
+        }
+    }
+
+    public void processMove(MoveTuple move) {
+        logger.info("Do move: " + move);
+        endCalculatorChecks();
+        if (turn(move) && !isStalement()) {
+            calculateCheckMate();
+        }
+    }
+
+    private void endCalculatorChecks() {
+        if (moveManager.getTurnCounter() > 10) {
+            logger.info("Calculate game end...");
+            endCalculator.changeState(colorOnTurn(), chessModel.getBoardModel());
+            if (endCalculator.validateStalement()) {
+                chessModel.setStalement(true);
+            }
+        }
+    }
+
+    private void calculateCheckMate() {
+        logger.info("Calculate check mate...");
```

Comment:
- [ ] distinguish between endCalculatorChecks and calculateCheckMate logger info for better debugging